magma/generator.py

```diff
@@ -67,14 +67,21 @@ def _make_type(cls, *args, **kwargs):
     dct = DefineCircuitKind.__prepare__(name, bases)
     cls.__init__(dummy, *args, **kwargs)
     dct.update(dict(dummy.__dict__))
-    return DefineCircuitKind.__new__(cls, name, bases, dct)
+    # Use circuit bind for the instance,
+    # rather than inherited Generator2 bind
+    dct["bind"] = classmethod(DefineCircuitKind.bind)
+    circuit = DefineCircuitKind.__new__(cls, name, bases, dct)
+    for gen in cls.bind_generators:
+        gen.generate_bind(circuit, *args, **kwargs)
+    return circuit
```

Comment:
```suggestion
    # NOTE(leonardt): We need to override the Generator2 classmethod bind with
    DefineCircuitKind.bind for generator instances (circuits).
    dct["bind"] = classmethod(DefineCircuitKind.bind)
    ckt = DefineCircuitKind.__new__(cls, name, bases, dct)
    for gen in cls.bind_generators:
        gen.generate_bind(ckt, *args, **kwargs)
    return ckt
```

Reply:
Last question: So the `generate_bind` method needs to have the same parameters as the bound generator? I ask because we pass `*args, **kwargs` down to `generate_bind`.

Reply:
The interface is "generated circuit, *<generator args>, **<generator kwargs>", so you the first argument is the result from the generator (so you have a handle to it for calling bind, as well as accessing internal signals to bind to).  Then, yes, the rest of the arguments should be the same as generator arguments (this is so the bind generator has a handle to them in case they want to use them in the code gen).

Reply:
This was added before we had Generator2 (hence the migration), I wonder if we could drum up something a little better that fits into the Generator2 style (e.g. this uses the staticmethod style that Generator had)